multiqc/modules/samtools/stats.py

```diff
@@ -10,125 +10,163 @@
+            keys['reads_mapped_and_paired'] = dict(reads, **{'title': 'Mapped &amp; paired', 'description': 'Paired-end technology bit set + both mates mapped' })
+            keys['reads_properly_paired'] = dict(reads, **{'title': 'Properly paired', 'description': 'Proper-pair bit set' })
+            keys['reads_duplicated'] = dict(reads, **{'title': 'Duplicated', 'description': 'PCR or optical duplicate bit set' })
+            keys['reads_QC_failed'] = dict(reads, **{'title': 'QC Failed'})
+            keys['reads_MQ0'] = dict(reads, **{'title': 'Reads MQ0', 'description': 'Reads mapped and MQ=0' })
+            keys['bases_mapped_(cigar)'] = dict(bases, **{'title': 'Mapped bases (cigar)', 'description': 'Mapped bases (cigar)' })
+            keys['bases_trimmed'] = dict(bases, **{'title': 'Bases Trimmed' })
+            keys['bases_duplicated'] = dict(bases, **{'title': 'Duplicated bases' })
+            keys['pairs_on_different_chromosomes'] = dict(reads, **{'title': 'Diff chromosomes', 'description': 'Pairs on different chromosomes' })
+            keys['pairs_with_other_orientation'] = dict(reads, **{'title': 'Other orientation', 'description': 'Pairs with other orientation' })
+            keys['inward_oriented_pairs'] = dict(reads, **{'title': 'Inward pairs', 'description': 'Inward oriented pairs' })
+            keys['outward_oriented_pairs'] = dict(reads, **{'title': 'Outward pairs', 'description': 'Outward oriented pairs' })
+
+            self.sections.append({
+                'name': 'Samtools Stats',
+                'anchor': 'samtools-stats',
+                'content': '<p>This module parses the output from <code>samtools stats</code>. All numbers in millions.</p>' +
+                            plots.beeswarm.plot(self.samtools_stats, keys, {'id': 'samtools-stats-dp'})
+            })
         
-        self.sections.append({
-            'name': 'Samtools Stats',
-            'anchor': 'samtools-stats',
-            'content': '<p>This module parses the output from <code>samtools stats</code>. All numbers in millions.</p>' +
-                        plots.beeswarm.plot(self.samtools_stats, keys, {'id': 'samtools-stats-dp'})
-        })
-    
-    # Return the number of logs that were found
-    return len(self.samtools_stats)
+        # Return the number of logs that were found
+        return len(self.samtools_stats)
+
+
+def alignment_section(samples_data):
+    bedgraph_data = {}
+    for sample_id, data in samples_data.items():
+        expected_total = data['raw_total_sequences']
+        read_sum = (data['reads_mapped'] + data['reads_unmapped'])
+        if read_sum == expected_total:
+            bedgraph_data[sample_id] = data
+        else:
+            log.warn("sum of mapped/unmapped reads not matching total, "
+                     "skipping samtools plot for: {}".format(sample_id))
+    bargraph = alignment_chart(bedgraph_data)
```

Comment:
oh, it's just a preference of mine

every function I try to keep to doing one thing. I know testing is a ways a way but if I'd write tests for this code it would make it easier to have this separate since it doesn't interact with the rest of the class :)

But it's good to unify code structure across a project so tell me if you want me to update!